### PR TITLE
topolog2.0: add volume support for ipc4

### PR DESCRIPTION
**Pull request by RanderWang**
volume is supported by gain module

### CODE REVIEW

File: tools/topology/topology2/include/components/virtual_widget.conf [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#
+# SOF virtual widget
+#
+# A generic volume component. All attributes defined herein are namespaced
```
Comment (lgirdwood) [inline tools/topology/topology2/include/components/virtual_widget.conf:4]: This looks like a copy paste from volume component. Can you add the correct comment for virtual widget - I assume it maps to the ASoC virtual widget ? if so , we should say so in the comments.

File: tools/topology/topology2/include/controls/bytes.conf [hunk 2]
```diff
@@ -0,0 +1,72 @@
+<include/controls/common.conf>
+
```
Comment (lgirdwood) [inline tools/topology/topology2/include/controls/bytes.conf:2]: Can you include an example instantiation in the comments here like the others have.

**Review by lgirdwood — COMMENTED** [review]
@ranj063 any comments ?

File: tools/topology/topology2/include/components/virtual_widget.conf (hunk 1 above)
Comment (RanderWang) [reply tools/topology/topology2/include/components/virtual_widget.conf:4]: thanks

File: tools/topology/topology2/include/controls/bytes.conf (hunk 2 above)
Comment (RanderWang) [reply tools/topology/topology2/include/controls/bytes.conf:16]: sure

**Comment by RanderWang** [comment]
updated. thanks!

File: tools/topology/topology2/include/components/virtual_widget.conf [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+# by alsatplg to "pga.attribute_name"
```
Comment (ranj063) [inline tools/topology/topology2/include/components/virtual_widget.conf]: @RanderWang still says volume here in the comment. Can you please change to virtual_widget?

File: tools/topology/topology2/include/dais/hda.conf [hunk 4]
```diff
@@ -0,0 +1,85 @@
+#
+# HDA DAI
+#
+# All attributes defined herein are namespaced by alsatplg to "Object.Dai.hda.attribute_name"
+#
+# Usage: this component can be used by declaring as follows:
+#
+# For Playback
+#Object.HDA."N" {
```
Comment (ranj063) [inline tools/topology/topology2/include/dais/hda.conf:9]: @RanderWang can you please fix the comments here? to show the right usage? It should be Object.Dai.HDA.N

File: tools/topology/topology2/include/dais/hda.conf [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+#		direction	"playback"
+#		stream_name	"iDisp1"
+#		id		2
+#		Object.Base.hw_config."0" {}
+# }
+# For Capture
+#Object.HDA."N" {
```
Comment (ranj063) [inline tools/topology/topology2/include/dais/hda.conf:16]: same here

File: tools/topology/topology2/include/pipelines/cavs/volume-capture.conf [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+# 	period		1000
+# 	time_domain	"timer"
+# 	channels	2
+# 	rate		48000
+# }
+#
+# Where N is the unique pipeline ID within the same alsaconf node.
+#
+
+Class.Pipeline."volume-capture" {
```
Comment (ranj063) [inline tools/topology/topology2/include/pipelines/cavs/volume-capture.conf:20]: @RanderWang this name will clash with the volume-cpature we use in SOF no? Can we use a different name

File: tools/topology/topology2/include/pipelines/cavs/volume-playback.conf [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+# 	period		1000
+# 	time_domain	"timer"
+# 	channels	2
+# 	rate		48000
+# }
+#
+# Where N is the unique pipeline ID within the same alsaconf node.
+#
+
+Class.Pipeline."volume-playback" {
```
Comment (ranj063) [inline tools/topology/topology2/include/pipelines/cavs/volume-playback.conf:20]: same with this one too

File: tools/topology/topology2/include/pipelines/cavs/volume-capture.conf (hunk 6 above)
Comment (RanderWang) [reply tools/topology/topology2/include/pipelines/cavs/volume-capture.conf:20]: yes. actually there is no module called volume in ipc4 and it is gain module. It is named to volume to let us easy to get it.  Maybe gain is more suitable for ipc4

File: tools/topology/topology2/include/components/virtual_widget.conf (hunk 3 above)
Comment (RanderWang) [reply tools/topology/topology2/include/components/virtual_widget.conf]: sorry,  I merged the change into another commit. I will fix it.

File: tools/topology/topology2/include/dais/hda.conf (hunk 4 above)
Comment (RanderWang) [reply tools/topology/topology2/include/dais/hda.conf:9]: sure

File: tools/topology/topology2/include/pipelines/cavs/volume-playback.conf (hunk 7 above)
Comment (RanderWang) [reply tools/topology/topology2/include/pipelines/cavs/volume-playback.conf:20]: sure

File: tools/topology/topology2/include/dais/hda.conf (hunk 5 above)
Comment (RanderWang) [reply tools/topology/topology2/include/dais/hda.conf:16]: sure

**Comment by RanderWang** [comment]
thanks, update my PR.

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**